src/integrationTest/java/org/opensearch/security/privileges/PrivilegesEvaluatorTest.java

```diff
@@ -44,10 +45,15 @@ public class PrivilegesEvaluatorTest {
         new Role("negated_regex_role").indexPermissions("read").on("/^[a-z].*/").clusterPermissions("cluster_composite_ops")
     );
 
+    protected final static TestSecurityConfig.User SEARCH_TEMPLATE = new TestSecurityConfig.User("search_template_user").roles(
+        new Role("search_template_role").indexPermissions("read").on("/^[a-z].*/").clusterPermissions("indices:data/read/search/template")
```

Comment:
Thank you @derek-ho ! This change looks good to me. I only have one concern about breaking existing roles that are unknowingly working with the bug in place, but since it is a bug fix I think its alright to provide documentation on how to update a role to make it so that SearchTemplateRequests are working properly. I also think it is worthwhile to modify the [`cluster_composite_ops_ro` static action group](https://github.com/opensearch-project/security/blob/main/src/main/resources/static_config/static_action_groups.yml#L121-L134) to include search template and msearch template.

To be clear, the role below allows SearchTemplateRequests to work even with the bug in place:

```
search_all:
  index_permissions:
    - index_patterns:
        - "*"
      allowed_actions:
        - "read"
```

and after this change the role would need to be changed to:

```
search_all:
  cluster_permissions:
    - "indices:data/read/search/template" or "cluster_composite_ops_ro" <- action group containing the underlying cluster perm
  index_permissions:
    - index_patterns:
        - "*"
      allowed_actions:
        - "read"
```

With the bug in place the following role would not permit search template requests to work properly

```
logs_search:
  index_permissions:
    - index_patterns:
        - "logs-*"
      allowed_actions:
        - "read"
```

but that would be achievable now with the following role definition:

```
logs_search:
  cluster_permissions:
    - "indices:data/read/search/template"
  index_permissions:
    - index_patterns:
        - "logs-*"
      allowed_actions:
        - "read"
```

Reply:
I don't think we can make the change for 2.x since it will be a breaking change. While it is an edge case, I don't think it matters if it is for a bug fix or not. Unless I am mistaken, semantic versioning would dictate we wait until 3.0 to make this available.

Reply:
Can you remove the `clusterPermissions` section for this role?